File: LeetCode/104/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def traverse(root, depth) :
            if(root) :
                return max(traverse(root.left, depth+1), traverse(root.right, depth+1))
            else :
                return depth
            
        return traverse(root, 0)

File: LeetCode/104/main.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
  int md(TreeNode *node, int height) {
    if (node == nullptr) {
      return height;
    }
    int left = md(node->left, height + 1);
    int right = md(node->right, height + 1);
    return max(left, right);
  }

  int maxDepth(TreeNode *root) { return md(root, 0); }
};